src/cssm.pyx

```diff
@@ -2537,6 +2537,385 @@ def ddm_flexbound_seq2(np.ndarray[float, ndim = 1] vh,
+            y_ha = (-1) * boundary_view[0] + (zh_view[k,0] * 2 * (boundary_view[0]))  # reset starting position 
+            y_hb = (-1) * boundary_view[0] + (zh_view[k,1] * 2 * (boundary_view[0])) 
+            if n == 0:
+                if k == 0:
+                    traja_view[0, 0] = y_ha
+                    trajb_view[0, 0] = y_hb
+
+            while y_ha <= boundary_view[ix] and y_hb <= boundary_view[ix] and t_particle <= deadline_tmp:
+                y_ha += (vha_view[k] * delta_t) + (sqrt_sta * gaussian_values[m])
+                y_ha = fmax(0.0, y_ha) 
+                m += 1
+                if m == num_draws:
+                    gaussian_values = draw_gaussian(num_draws)
+                    m = 0
+
+                y_hb += (vhb_view[k] * delta_t) + (sqrt_stb * gaussian_values[m])
+                y_hb = fmax(0.0, y_hb)
+                m += 1
+                if m == num_draws:
+                    gaussian_values = draw_gaussian(num_draws)
+                    m = 0
+                t_particle += delta_t
+                ix += 1
+               
+
+                if n == 0:
+                    if k == 0:
+                        traja_view[ix, 0] = y_ha
+                        trajb_view[ix, 0] = y_hb
+
+            # If we are already at maximum t, to generate a choice we just sample from a bernoulli
+            if t_particle >= max_t:
+                # High dim choice depends on position of particle
+                if boundary_view[ix] <= 0:
+                    if random_uniform() <= 0.5:
+                        choices_view[n, k, 0] += 2
+                elif random_uniform() <= ((y_ha + boundary_view[ix]) / (2 * boundary_view[ix])):
+                        choices_view[n, k, 0] += 2
+
+                # Low dim choice random (didn't even get to process it if rt is at max after first choice)
+                # so we just apply a priori bias
+                if choices_view[n, k, 0] == 0:
+                    if random_uniform() <= zl1_view[k,0]:
+                        choices_view[n, k, 0] += 1
+                else:
+                    if random_uniform() <= zl2_view[k,0]:
+                        choices_view[n, k, 0] += 1
+                rts_view[n, k, 0] = t_particle
+                decision_taken = 1
+            else:
+                y_h = fmax(y_ha, y_hb)
+                # If boundary is negative (or 0) already, we flip a coin
+                if boundary_view[ix] <= 0:
+                    if random_uniform() <= 0.5:
+                        choices_view[n, k, 0] += 2
+                # Otherwise apply rule from above
+                elif y_ha >= boundary_view[ix]: #High dim racer wins
+                    choices_view[n, k, 0] += 2
+
+                # elif random_uniform() <= ((y_h + boundary_view[ix]) / (2 * boundary_view[ix])):
+                #     choices_view[n, k, 0] += 2
+
+                y_l1a = (-1) * boundary_view[ix] + (zl1_view[k,0] * 2 * (boundary_view[ix]))
```

Comment:
as explained below, the starting point here should take into account that the boundary is not `[-bound(t), bound(t)]` but rather `[0, bount(t)]`

Reply:
fixed